uplink/ecclient/client.go

```diff
@@ -148,6 +149,18 @@ func (ec *ecClient) Put(ctx context.Context, limits []*pb.AddressedOrderLimit, p
 	}()
 
 	successes := int(atomic.LoadInt32(&successfulCount))
+	mon.IntVal("segment_pieces").Observe(int64(pieceCount))
+	mon.FloatVal("segment_pieces_percent_success").Observe(float64(successes) / float64(rs.OptimalThreshold()))
+
+	var failures int
+	for range limits {
+		info := <-infos
```

Comment:
I don't think you can listen on the channel like this. We are already piping exactly one `info` into `infos` per `ec.putPiece` above, and those were already consumed. At this point in execution, all the `info`s should have already been received by the for loop above.

Reply:
Ah you're right. It's been a minute since I've used the channels. How embarrassing. I'll fix this.